src/cinema.ts

```diff
@@ -0,0 +1,172 @@
+// deno-lint-ignore-file no-explicit-any adjacent-overload-signatures
+import { Json } from './beatmap.ts';
+import { jsonGet } from './general.ts';
+
+export class CinemaScreen {
+    json: Record<string, any> = {
+        "videoID": undefined,
+        "screenPosition": {
+            "x": 0.0,
+            "y": 12.4,
+            "z": 67.8
+        },
+        "screenRotation": {
+            "x": -8.0,
+            "y": 0.0,
+            "z": 0.0
+        }
+    }
+
+    /**
+     * Cinema Screen object for ease of creation
+     * @param {CINEMA_LOOKUP} cinemaLookupMethod You can either play a video with Cinema by either using a YouTube video ID, or a video URL (YouTube, Facebook, Dailymotion, Vimeo)
+     * @param {String} videoID YouTube video ID or video URL
+     * @param {String} videoFile Name of the video file on the local file system.
+     */
+    constructor(cinemaLookupMethod: CINEMA_LOOKUP, videoID: string, videoFile?: string) {
+        if (cinemaLookupMethod == "URL" && cinemaLookupMethod !== undefined) this.videoUrl = videoID;
+        if (cinemaLookupMethod == "YoutubeID" && cinemaLookupMethod !== undefined) this.videoID = videoID;
+        if (videoFile !== undefined) this.videoFile = videoFile;
+    }
+
+    /**
+     * Create a Cinema Screen using Json.
+     * @param json Json to import.
+     */
+    import(json: Json) {
+        this.json = json;
+        return this;
+    }
+
+    /**
+     * Push Main Cinema Screen to cinema-video.json
+     * @returns 
+     */
+    push() {
+        // Warnings for out of range numbers for certain Cinema Properties
+        function cinemaRange(thing: string, number: number) {
+            console.log("\x1b[31m", `Cinema Warning: ${thing} value`, "\x1b[31m", number, "\x1b[31m", "out of range!",  "\x1b[0m")
```

Comment:
Any chance you could use constants/variables for the ASCII color coding for clarity?

Reply:
👍